alica_engine/src/engine/scheduler/JobQueue.cpp

```diff
@@ -0,0 +1,59 @@
+#include "engine/scheduler/JobQueue.h"
+
+namespace alica {
+namespace scheduler {
+    void JobQueue::insert(std::shared_ptr<Job> job)
+    {
+        {
+            std::unique_lock<std::mutex> lock(_mtx);
+            _queue.push_back(job);
+            std::sort(_queue.begin(), _queue.end());
```

Comment:
Do not sort a whole vector when you just want to insert one element. Insert the element at the right position.
`std::upper_bound` is your friend.